### PR TITLE
Loading to staging schemas db from database-specific directory

### PR DESCRIPTION
**Pull request by srinathgit**
Loading to staging schemas db from database-specific directory

This PR

1. Scaffolds ml-config/databases/<staging-db-name>/schemas for the purpose of loading schemas and stops scaffolding the existing hub-internal-config/schemas dir
2. Makes the same change to the test
3. Fixes another test (where triggers count was incorrect)
4. Updates ml-gradle and ml-app-deployer to 3.11.0 and java client to 4.1.2

### CODE REVIEW

**Comment by srinathgit** [comment]
The last commit https://github.com/marklogic/marklogic-data-hub/pull/1768/commits/96163ba7bb81ee5ed77b0b5ce5db19dbb0dcf505 is not related to the title of PR. It fixes 'LoadUserArtifactsCommand' and some of the test failures

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java [hunk 1]
```diff
@@ -105,7 +106,11 @@ public void createProject(String projectDirString) {
         return getHubConfigDir().resolve("security");
     }
 
-    @Override public Path getHubSchemasDir() { return getHubConfigDir().resolve("schemas"); }
+    @Override public Path getHubSchemasDir() { return this.stagingSchemasDir; }
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java]: I would remove this, along with the private stagingSchemasDir variable. I don't think there's any reason for it - other than perhaps backwards compatibility, but I don't think that's necessary in this case because a feature is being removed - i.e. there's no need to keep track of this.

File: ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/DataHubPlugin.groovy [hunk 2]
```diff
@@ -118,11 +118,13 @@ class DataHubPlugin implements Plugin<Project> {
...
 
         // This isn't likely to be used, but it's being kept for regression purposes for now
-        project.task("hubDeployUserModules", group: deployGroup, type: DeployUserModulesTask, description: "Installs user modules from the plugins and src/main/entity-config directories.")
+        project.task("hubDeployUserModules", group: deployGroup, type: DeployUserModulesTask,
+            description: "Installs user modules from the plugins and src/main/entity-config directories.")
+            .finalizedBy(["hubDeployUserArtifacts"])
 
         project.task("hubDeployUserArtifacts", group: deployGroup, type: DeployUserArtifactsTask,
             description: "Installs user artifacts such as entities and mappings.")
-            .mustRunAfter(["hubDeployUserModules"])
```
Comment (rjrudin) [inline ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/DataHubPlugin.groovy]: This looks like it should be in a separate PR as well.

**Comment by rjrudin** [comment]
I also recommend removing updateSchemaCommandList from DataHubImpl and providing an upgrade step (manual or automated) that will move anything under hub-internal-config/schemas to ml-config/databases/(schemas db name)/schemas. Should get rid of all of the extra code now.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java (hunk 1 above)
Comment (srinathgit) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java]: @aebadirad , 
Do you want to retain the getHubSchemasDIr() for backward compatibility for 4.2 release?

Comment (aebadirad) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java]: Yes. Mark it deprecated though.

**Comment by srinathgit** [comment]
Closing this , opened PR #1780
